Interrim Research Rework: Redemption

Sun-Soaked opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
A fairly simple redistribution of research tools, intended to limit the power of generalists and improve the RND experience of specialists. This isn't our final research form, which will likely be in the flavour of scavenging- this is just testing the waters of limited access and exclusivity.

This PR:
Creates specialized R&D Kits, which are braked to print specific subtypes of goods.
Grants every vessel an R&D Kit befitting it's specialization, or lack thereof.
Switches out existing omni-rnd kits in certain ships for generalist kits without any specialized equipment.
Slightly tweaks some printing requirements, adding a number of items to the "generalist" category that everyone can print.
Makes genetics research slightly cheaper, and folds out cloning into a follow-up tech.
Adds four new Satchel of Holding subtypes, for botany, construction, viro/xenobio, and chemistry.
Adds (ports from tegu) quantum components, which are rare hyper-efficient parts

Thanks to @Ms-Mee  for the fix to repeatable dissections!
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Specialists having access to exclusive technologies should boost inter-ship interaction considerably, leading to inter-ship trading/competition/theft.
This'll let us see how players work with different ships having different levels of opportunity and more obvious specialization paths, and let us see how they adapt to not being able to do everything themselves, particularly on vessels like the airpod shetty where roundstart RND sweeps became a bit common.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: allows exotic plant samples and slime cores to be redeemed for research points.
tweak: buffs survey back into light viability, but it also depletes far more quickly.
add: ports quantum parts, to be used as a high value loot-only part tier that's not typically constructible.
add: "department" specific versions of the circuit imprinter.
add: R&D boxes flavored to specific subtypes of research.
add: Many subtype RND boxes, to many ships.
add: four new satchels of holding, with sprites each based on an adjusted version of whitesand's Mining Satchel of Holding.
add: An expensive standard firing pin research, after advanced weapons tech.
tweak: replaced omni-RND on most ships that have it with specialized versions fitting the vessel's theme. 
tweak: Adds research equipment to many ships lacking it, in the form of the limited production sets above.
tweak: adjusts the department requirements of a number of tech items to support the nonspecialized RnD equipment.
balance: folded out cloning into it's own research, behind genetics.
fix: Prevents dissections from being repeated at every tech level. Dissections now give very little points on "common" races, and significantly more on rarer lifeforms.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sun-Soaked committed:
reebok shoes

-adds research construction boxes to all vessels currently missing them, albiet in the form of limited production subtypes
-switches out omniresearch equipment on many ships for limited-production versions
-respecs some departmental construction requirements
-redistributes some research slightly
-adds limited production "departmental" circuit imprinters
-adds themed subtype rnd boxes

Sun-Soaked committed:
robocop is a movie released in 1987

-makes a refernce slightly more obvious, on request

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
 	new /obj/item/circuitboard/computer/rdconsole(src)
 
+/obj/item/storage/box/rndboards/old
+	name = "\proper Nanotransen RND Construction Kit"
```
```suggestion
	name = "\proper Nanotrasen RND Construction Kit"
```
also, shouldn't it be R&D?

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
 
+/obj/item/storage/box/rndboards/old
+	name = "\proper Nanotransen RND Construction Kit"
+	desc = "A set of boards for constructing prototype design lathes, dating from a prewar Nanotransen labratory. These ones are unbraked, and can produce any of the designs in their database without limit."
```
```suggestion
	desc = "A set of boards for constructing prototype design lathes, dating from a prewar Nanotrasen laboratory. These ones are unbraked, and can produce any of the designs in their database without limit."
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+
+/obj/item/storage/box/rndengi
+	name = "\proper QWIK-RND: A.T.L.A.S. Module"
+	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print maintnence, construction, and repair related designs."
```
```suggestion
	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print maintenance, construction, and repair related designs."
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+
+/obj/item/storage/box/rndsec
+	name = "\proper QWICK-RND: P.E.A.C.E. Module"
+	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print low-to-midrange military designs."
```
`low-to-midrange military`
maybe `civilian` or `surplus` combat/weapon designs?

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+	new /obj/item/circuitboard/computer/rdconsole(src)
+
+/obj/item/storage/box/rndsec
+	name = "\proper QWICK-RND: P.E.A.C.E. Module"
```
```suggestion
	name = "\proper QWIK-RND: P.E.A.C.E. Module"
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+
+/obj/item/storage/box/rndbasic
+	name = "\proper QWICK-RND: B.A.S.I.C. Module"
+	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print a variety of low-tier miscallaneous designs."
```
```suggestion
	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print a variety of low-tier miscellaneous designs."
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+	new /obj/item/circuitboard/computer/rdconsole(src)
+
+/obj/item/storage/box/rndbasic
+	name = "\proper QWICK-RND: B.A.S.I.C. Module"
```
```suggestion
	name = "\proper QWIK-RND: B.A.S.I.C. Module"
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+	new /obj/item/circuitboard/computer/rdconsole(src)
+
+/obj/item/storage/box/rndciv
+	name = "\proper QWICK-RND: H.O.M.E. Module"
```
```suggestion
	name = "\proper QICK-RND: H.O.M.E. Module"
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+	new /obj/item/circuitboard/computer/rdconsole(src)
+
+/obj/item/storage/box/rndsci
+	name = "\proper QWICK-RND: K.N.O.W. Module"
```
```suggestion
	name = "\proper QWIK-RND: K.N.O.W. Module"
```

Firefox13 commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1155,6 +1155,88 @@
+
+/obj/item/storage/box/rndsci
+	name = "\proper QWICK-RND: K.N.O.W. Module"
+	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print desgins related to high-level scientific disciplines."
```
```suggestion
	desc = "A set of boards for constructing prototype design lathes. These ones are braked to only print designs related to high-level scientific disciplines."
```

Sun-Soaked committed:
guh

-fixes some spelling/grammer issues
-adds an expensive normal firing pin tech, branching from advanced weapons
-generalizes a number of printing options
-limits RCD creation to engineering, allows a number of types to print the RPD

Sun-Soaked committed:
forgot this

-generalizes smelting on protolathes

Sun-Soaked committed:
no more spunch bob

riggs 1 now has a science board focus

Sun-Soaked committed:
surgical moment

-more adjustments
-adds an RND choice beacon, implements it on the airpod shetty

untrius commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -474,11 +474,20 @@
+/datum/techweb_node/cloning
+	id = "cloning"
+	display_name = "Full-Body Reconstitution"
+	description = "We can rebuild him, we have the technologyx."
```
"technologyx"

Sun-Soaked replied:
:how:

Sun-Soaked committed:
:how:

-:how:

Sun-Soaked committed:
whoof

a

Sun-Soaked committed:
a

why does the cargo hold need these

Sun-Soaked committed:
Return of the King

-Survey is now significantly more robust, but depletes far more quickly

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
meework

-experimental dissection has been slightly reworked: Normal humans are now inefficient, but special entities are worth slightly more. Expands the list of special entities.
-slightly reduces the cost of a few early research items
-adds a two new species defines

Sun-Soaked committed:
whoops

guh

Sun-Soaked committed:
further tweaks and balances

-the funny surgery no longer becomes unstoppable

Sun-Soaked committed:
whooh

-adds server to boyardee

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
-liteguai tweaks

-replaces full RND set in crate with a circuit board.

Sun-Soaked committed:
GAK

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
fixes rigger

MarkSuckerberg committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Ms-Mee committed:
Should work

Sun-Soaked committed:
g

g

Sun-Soaked committed:
buffs

-g

Sun-Soaked committed:
Merge pull request #2 from Ms-Mee/patch-4

Should work

Sun-Soaked committed:
mistako

-mishakeo

Sun-Soaked committed:
-fix

fixes a syndicate crusher bug

Sun-Soaked committed:
EMERGENCY FIXES

fixes crusher AGAIN because I forgot how to write or apparently
fixes planet generation because I'm an idiot

Sun-Soaked committed:
spelling misnake

-hiss

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Ms-Mee committed:
Update experimental_dissection.dm

Ms-Mee committed:
Merge branch 'techfab' into patch-4

Ms-Mee committed:
Update code/modules/surgery/experimental_dissection.dm

Sun-Soaked committed:
Merge pull request #3 from Ms-Mee/patch-4

Fix v2

Sun-Soaked committed:
Swifter RND

-Nukes the 5 omnilathe kits(!!!) on the lux, replacing them with cargo sets
-Adds assorted RND kits to other swifter ships

Sun-Soaked committed:
Merge branch 'techfab' of https://github.com/MemedHams/Shiptest into techfab

Sun-Soaked committed:
a

s

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
-guh

-adds ashlizards to dissection
-basic Engimatic Part implementation.
-buffs upgraded dissections.

Sun-Soaked committed:
updated fix

-repairs the kinetic cleaver, turning it into a subtype
-fixes tendril snow legion subtype

Sun-Soaked committed:
storage balance destroyer

-adds chemical, plant, construction, and biohazard(Viro/xenobio) bags of holding.

Sun-Soaked committed:
xenostuff part 1

-moves slime researching from the xenobiology console to the R&D computer.
-Adds a staggering number of bolts to chaos magicarp, including a new one that functions as the opposite of a necropotence bolt. Embrace chaos!
-touches up the new bags of holding slightly

Sun-Soaked committed:
-changes

-adds slime and plant research to R&D consoles, allowing you to discover species of both for research points.
-buffs the wizar portal
-prevents mythril from presenting rpg effects on coins and stacks

Sun-Soaked committed:
no

-adds the unfinished slime guardian, fuck you travis

Sun-Soaked committed:
M!

-fixes the plant dna manipulator being unaqirable

MarkSuckerberg commented on code/datums/materials/basemats.dm:
```diff
@@ -253,12 +253,14 @@ Unless you know what you're doing, only use the first three numbers. They're in
 	. = ..()
 	if(istype(source, /obj/item))
-		source.AddComponent(/datum/component/fantasy)
+		if(!istype(source, /obj/item/stack/sheet || /obj/item/coin))
```
I don't think this works, also see the one on line 262

Sun-Soaked replied:
fixed(more or less)

Sun-Soaked committed:
-fixes

-fixes the dragonslayer spear and the legate

Sun-Soaked committed:
-BIG BOY

contains:
-wave 1 item modifications(unimplemented until I have sprites for them)
-weaker craftable and repeatable item sharpeners.
-fixes dragonslayer spear
-makes freeze cube fly half as fast, making it easier to dodge but also much easier to catch.
-being frozen in an ice cube now applies an extremely brief stun.
-adds the most advanced saw in the game, far surpassing any other weapon

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
demonic boost

-adds textures for modifiers, and implements them in necropolis/demonic pools.
-greatly boosts creature spawns and slightly boosts overall item chances, on the demonic portal pools
-replaces demonic portal wizards with a more cold resilient version
-adds a "the traps" spellbook

Sun-Soaked committed:
funky

Sun-Soaked committed:
Update hivelord.dm

fixes the bad no good stinky very bad base subtype issue

Sun-Soaked committed:
fix?

-please fix the attack chain, thanks

Sun-Soaked committed:
maybe?

hm

Sun-Soaked committed:
this will break something

Sun-Soaked committed:
check two

Sun-Soaked committed:
temporary state of events

Sun-Soaked committed:
-e

-this might work!

Sun-Soaked committed:
stayin alive, stayin alive

-emergency fixes to critical cooldown errors
-emergency fixes to alt-click errors
-polishes up roughness with the The Traps spellbook sprite

Sun-Soaked committed:
FUCK

-nerfs zephyr

-stupid stupid stupid stupid stupid

Sun-Soaked committed:
m

Sun-Soaked committed:
m

-final emergency tweaks

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
holohaira

-gives the healing holopara a stand cry when in combat mode, boosts damage and healing slightly
-makes the explosive effect actually a chance instead of triggering on every hit, causes it to slightly ignite targets

Sun-Soaked committed:
WHOOPS

-fix
-also boosts standard holoparasite damage against simplemobs

Sun-Soaked committed:
3 am coding

MarkSuckerberg commented on code/game/objects/items.dm:
```diff
@@ -205,8 +237,13 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 		m.temporarilyRemoveItemFromInventory(src, TRUE)
 	for(var/X in actions)
 		qdel(X)
+	if(possessed == TRUE)
```
```suggestion
	if(possessed)
```

MarkSuckerberg commented on code/game/objects/items.dm:
```diff
@@ -278,7 +315,18 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 		if(resistance_flags & FIRE_PROOF)
 			. += "[src] is made of fire-retardant materials."
-
+//modifier annendums
```
```suggestion
//modifier addendums
```

MarkSuckerberg commented on code/game/objects/items.dm:
```diff
@@ -172,6 +172,35 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
+
+//causes an effect overlay to appear over your char when the item is held inhand. Most often used for "shielding" effects, see red and blue holy staves for an example.
+	var/overlayed = FALSE
+	var/effect_icon = null//pathing is under icons/effects/effects.dmi
```
I think some of these booleans should be relegated to item flags, if possible.

Sun-Soaked replied:
I'll investigate doing this as part of the eris item mod port

MarkSuckerberg reviewed (changes requested):
Everything else seems to be in order

Sun-Soaked committed:
When the review is sus!

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
Update code/game/objects/items.dm

whoops

Co-authored-by: Mark Suckerberg <29362068+MarkSuckerberg@users.noreply.github.com>

Sun-Soaked committed:
Update code/game/objects/items.dm

m

Co-authored-by: Mark Suckerberg <29362068+MarkSuckerberg@users.noreply.github.com>

Sun-Soaked committed:
m...

Sun-Soaked committed:
Merge branch 'techfab' of https://github.com/MemedHams/Shiptest into techfab

MarkSuckerberg committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
Merge branch 'techfab' of https://github.com/MemedHams/Shiptest into techfab

Sun-Soaked committed:
woo yeah?

-fixes 3 drops per chest bug

MarkSuckerberg committed:
Removes (mil)synth checks from exp. dissection

ZephyrTFA commented on code/modules/research/stock_parts.dm:
```diff
@@ -328,3 +328,94 @@ If you create T5+ please take a pass at gene_modder.dm [L40]. Max_values MUST fi
+
+
+//Mysterious machine components made by an unknown source. Designed to be doled out as an ultra-rare loot item.
+/obj/item/stock_parts/capacitor/quantumcap
```
Why is this not documented in any capacity.

ZephyrTFA replied:
This is being marked as DNM until I have the time to go over this entire PR

Sun-Soaked replied:
they're not currently aquirable ingame (outside of a rare portal drop, as common) but I'm planning to use them as part of salvaging 2.
It seems like I failed to save the PR changelog changes with quantum, though

Sun-Soaked replied:
added to the main PR body

Sun-Soaked committed:
m!

Sun-Soaked committed:
please

hampter

triplezeta commented on code/__DEFINES/status_effects.dm:
```diff
@@ -72,6 +72,9 @@
 
+#define STATUS_EFFECT_DEEPFREEZE /datum/status_effect/stacking/freeze //if the chill builds up enough, takes a bit of damage and get frozen inside a block of ice
+
+#define STATUS_EFFECT_CUBEFREEZE /datum/status_effect/ice_block_talisman //creates a block of ice around someone, briefly stunning them and temporarily preventing them from moving.
```
newline here?
```suggestion
#define STATUS_EFFECT_CUBEFREEZE /datum/status_effect/ice_block_talisman //creates a block of ice around someone, briefly stunning them and temporarily preventing them from moving.

```

triplezeta commented on code/game/objects/items/sharpener.dm:
```diff
@@ -4,53 +4,317 @@
+	var/toolincrement = 0//subtract by decimal percentages for best effect, if trying to upgrade
 	var/max = 30
-	var/prefix = "sharpened"
+	var/prefix = "sharpen"
```
"That's a sharpen knife"

```suggestion
	var/prefix = "sharpened"
```

triplezeta commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -90,6 +90,252 @@
+
+/obj/structure/closet/crate/necropolis/tendril/greater/PopulateContents()
+	var/loot = rand(1,32)
+	switch(loot)
```
having 3 switch statements seems like the worst possible way to do this.

Sun-Soaked replied:
it is, and I was planning to ask Marg to walk me through making a loop before the whole VSC debacle happened.

triplezeta replied:
i see

Sun-Soaked replied:
we'll continue this on the next episode of interrim z, when I have to fix this in the mining pr

triplezeta commented on code/modules/mob/living/simple_animal/hostile/jungle/seedling.dm:
```diff
@@ -30,6 +30,7 @@
 	robust_searching = TRUE
 	stat_attack = HARD_CRIT
 	move_resist = MOVE_FORCE_EXTREMELY_STRONG
+	gold_core_spawnable = HOSTILE_SPAWN
```
🤨

ZephyrTFA replied:
This is both unfair and unfun to fight against unless you have TK and a very good melee weapon; probably not a good idea.

Sun-Soaked committed:
Update code/game/objects/items/sharpener.dm

Co-authored-by: tetra zeta  <51932018+triplezeta@users.noreply.github.com>

Sun-Soaked committed:
Update code/__DEFINES/status_effects.dm

Co-authored-by: tetra zeta  <51932018+triplezeta@users.noreply.github.com>

ZephyrTFA commented on code/__DEFINES/research.dm:
```diff
@@ -77,11 +78,19 @@
-#define SLIME_RESEARCH_TIER_4 2000
-#define SLIME_RESEARCH_TIER_5 2500
-#define SLIME_RESEARCH_TIER_RAINBOW 5000
+///Research point values for slime extracts, see the R&D console's attackby code
+#define SLIME_RESEARCH_TIER_0 500
+#define SLIME_RESEARCH_TIER_1 1500
+#define SLIME_RESEARCH_TIER_2 3000
+#define SLIME_RESEARCH_TIER_3 6000
+#define SLIME_RESEARCH_TIER_4 9000
+#define SLIME_RESEARCH_TIER_5 15000
+#define SLIME_RESEARCH_TIER_RAINBOW 20000
```
This seems like a very large amount of points to give with how many slime extracts you can get quickly.

Sun-Soaked replied:
each slime extract can only be redeemed once.

ZephyrTFA commented on code/__DEFINES/research.dm:
```diff
@@ -77,11 +78,19 @@
+#define SLIME_RESEARCH_TIER_5 15000
+#define SLIME_RESEARCH_TIER_RAINBOW 20000
+
+///Research point values for plant discoveries
+#define PLANT_RESEARCH_TIER_0 500
+#define PLANT_RESEARCH_TIER_1 1500
+#define PLANT_RESEARCH_TIER_2 3000
+#define PLANT_RESEARCH_TIER_3 6000
+#define PLANT_RESEARCH_TIER_4 9000
+#define PLANT_RESEARCH_TIER_5 15000
```
I'm not sure how plant discoveries work, if they are only once per techweb I'm fine with this.

Sun-Soaked replied:
both plant discoveries and slime extracts are "discoveries" that can only be redeemed once. That being said, I do need help connecting their "discovered items" lists to ships directly, so that you can't do the old reconstruct the console trick to just redeem everything again

ZephyrTFA commented on code/_onclick/item_attack.dm:
```diff
@@ -109,6 +109,38 @@
+		if(!B)
+			M.apply_status_effect(STATUS_EFFECT_SAWBLEED, bleedpower)
+		else
+			B.add_stacks(bleedpower)
```
This is a lot of copy and paste code, consider turning it into a macro.

Sun-Soaked replied:
atomized out.

ZephyrTFA commented on code/datums/action.dm:
```diff
@@ -720,6 +720,10 @@
 
+/datum/action/small_sprite/megafauna/marg
+	small_icon = 'icons/obj/plushes.dmi'
+	small_icon_state = "plushie_lizard"
```
atomise this.

ZephyrTFA commented on code/datums/status_effects/debuffs.dm:
```diff
@@ -353,8 +353,8 @@
-	tick_interval = 6
-	delay_before_decay = 5
+	tick_interval = 20
+	delay_before_decay = 20
```
atomsize

ZephyrTFA commented on code/datums/status_effects/debuffs.dm:
```diff
@@ -380,6 +380,29 @@
+	owner.apply_status_effect(STATUS_EFFECT_CUBEFREEZE, 6 SECONDS)//finally, recursive status datums. The future is now
+	var/turf/T = get_turf(owner)
+	new /obj/effect/temp_visual/freeze/explode(T)
+	playsound(T, 'sound/effects/glassbr3.ogg', 100, TRUE, -1)
```
You have far too many changes in one PR. Atomize.

ZephyrTFA commented on code/game/objects/items.dm:
```diff
@@ -172,6 +172,35 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
+	var/possessed = FALSE
+	//modularizes bleed effect. Bleedpower dictates how quickly stacks are gained on attack.
+	var/serrated = FALSE
+	var/bleedpower = 1
```
Use autodoc commenting.
```
/// Description of var
var/concise_var_name
```

ZephyrTFA commented on code/game/objects/items.dm:
```diff
@@ -181,7 +210,10 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
+//	if(!haunted)
+		actions_types = null
+//	else
+//		actions_types = list(/datum/action/item_action/possession)
```
Atomize.
And don't comment out code if you no longer need it, remove it.

ZephyrTFA commented on code/game/objects/items/granters.dm:
```diff
@@ -273,6 +273,19 @@
+/obj/item/book/granter/spell/traps/recoil(mob/living/user)
+	..()
+	to_chat(user, "<span class='danger'><B>The ground quakes beneath your feet!</B></span>")
+	user.Paralyze(100)
```
Atomize.

ZephyrTFA commented on code/game/objects/items/sharpener.dm:
```diff
@@ -4,53 +4,317 @@
+	var/stackable = FALSE//for effects that can be applied multiple times, set to TRUE
+	var/modify_text = "sharpen"//used in the flavour when sharpening items
+	var/use_noise = 'sound/items/unsheath.ogg'
+	var/postfix = null//allows special modifiers to give an RPG-style postfix instead, helps with name crowding
```
Atomize.

ZephyrTFA commented on code/game/objects/items/sharpener.dm:
```diff
@@ -4,53 +4,317 @@
 		to_chat(user, "<span class='warning'>You can only sharpen items that are already sharp, such as knives!</span>")
 		return
-	if(istype(I, /obj/item/melee/transforming/energy))
+	if(istype(I, /obj/item/melee/transforming/energy))//no esword sharpening
```
Unneeded comment.

ZephyrTFA commented on code/game/objects/items/sharpener.dm:
```diff
@@ -4,53 +4,317 @@
+		I.overlayed = TRUE
+		I.effect_icon = "shield-old"
+		I.AddComponent(/datum/component/anti_magic, TRUE, TRUE, FALSE, null, null, FALSE)
+	return
```
All of these sharpener changes need to be in a separate PR. (Atomize)

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -158,72 +158,113 @@ GLOBAL_LIST_INIT(ore_probability, list(
 				new /mob/living/simple_animal/hostile/clown/clownhulk(loc)
-			new /mob/living/simple_animal/hostile/clown/longface(loc)
+				new /mob/living/simple_animal/hostile/clown/longface(loc)
+				new /obj/item/shield/energy/bananium(loc)
```
No.

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -158,72 +158,113 @@ GLOBAL_LIST_INIT(ore_probability, list(
+				new /mob/living/simple_animal/hostile/clown/honkling(loc)
+			if(prob(25))
 				new /mob/living/simple_animal/hostile/clown/clownhulk/chlown(loc)
+				new /obj/item/shield/energy/bananium(loc)
```
No.

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -158,72 +158,113 @@ GLOBAL_LIST_INIT(ore_probability, list(
-			if(prob(25))
+				new /mob/living/simple_animal/hostile/clown/fleshclown(loc)
+			if(prob(35))
+				new /obj/item/clothing/suit/space/hardsuit/clown/combat(loc)
```
No.

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -232,15 +273,19 @@ GLOBAL_LIST_INIT(ore_probability, list(
-				new /obj/item/storage/box/holy_grenades(loc)
 				new /mob/living/simple_animal/hostile/skeleton/templar(loc)
 			if(prob(35))
+				new /obj/item/storage/box/holy_grenades(loc)
```
Questionable, but not against it.

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -914,13 +1075,27 @@ GLOBAL_LIST_INIT(ore_probability, list(
 				new /mob/living/simple_animal/hostile/zombie(loc)
+			if(prob(15))
+				new /obj/effect/spawner/lootdrop/engimatic(loc)
+				new /mob/living/simple_animal/hostile/zombie(loc)
```
Arent these the zombies that can spread Romerol?

ZephyrTFA commented on code/modules/events/wizard/magicarp.dm:
```diff
@@ -63,10 +64,13 @@
+	allowed_projectile_types = list(/obj/projectile/magic/change, /obj/projectile/magic/animate, /obj/projectile/magic/resurrection,
+	/obj/projectile/magic/death, /obj/projectile/magic/teleport, /obj/projectile/magic/door, /obj/projectile/magic/aoe/fireball,
+	/obj/projectile/magic/spellblade, /obj/projectile/magic/arcane_barrage, /obj/projectile/magic/locker, /obj/projectile/magic/nothing,
+	/obj/projectile/magic/aoe/lightning, /obj/projectile/magic/necropotence, /obj/projectile/magic/fortify, /obj/projectile/magic/spell/magic_missile)//now THIS is chaos
```
Atomize.

Sun-Soaked replied:
done

ZephyrTFA commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -90,6 +90,252 @@
+		if(1)
+			new /obj/item/shared_storage/red(src)
+		if(2)
+			new /obj/item/clothing/suit/space/hardsuit/cult(src)
```
Given how unlikely it is that this rolls, I'm fine with it.

ZephyrTFA commented on code/modules/mob/living/simple_animal/hostile/alien.dm:
```diff
@@ -30,7 +30,7 @@
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	unique_name = 1
-	gold_core_spawnable = NO_SPAWN
+	gold_core_spawnable = HOSTILE_SPAWN
```
Terrible idea, but you know what you are getting into when you use gold cores so I approve.

Sun-Soaked replied:
:joker:

ZephyrTFA commented on code/modules/projectiles/projectile/magic.dm:
```diff
@@ -558,6 +558,33 @@
+			spell.charge_counter = spell.charge_max
+			spell.recharging = FALSE
+			spell.update_icon()
+
```
Atomize.

ZephyrTFA reviewed (changes requested):
There were enough files that my browser was beginning to stall and crash the longer I reviewed.
Main issues are:
 - Atomization
 - Seemingly random power creep for loot spawning
 - Power creep for point generation

The reason behind me very strictly enforcing atomization is that in the future if someone fucks up or goes wrong we need to be able to narrow it down quickly to resolve issues.
If all of your changes are in this singular PR and merging it or TMing it causes major issues, I would be required to revert the entirety of this PR, which you will then make another PR to fix it and get it remerged. This causes endless frustration and needless work from both sides.
Keep this PR about reworking Research, or keep it about reworking Mining; don't do both in the same PR.
There is also the fact that if you atomize your changes your PR will be able to get reviewed much faster, which will allow a faster TM, which in turn will mean that you get data to bugfix and tweak earlier.

I understand that I seem like an asshole for enforcing these rules, but I enforce them for a reason; I do not want you, myself, or any of the other maintainers and coders to have to struggle with constant back and forth bug fixing and reviews.

If you have any questions you can DM me on Discord and I will try to make time to answer them.

Sun-Soaked committed:
removes seedling from hostile spawn pool

Sun-Soaked committed:
M!

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
m

m

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
adds the plant researching functionality

How did I forget to upload this

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
replaces the broken syndie crushers

I changed the directory, sorry

Sun-Soaked committed:
atomization pass 1

The End Of Time

Sun-Soaked committed:
e

Sun-Soaked committed:
e

new shiptest content merged in, gonna redo the interrim research changes here

Sun-Soaked committed:
please

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
l + death bolt + no polymorph + teleported into hyperspace

moves magicarp to mining pr

Sun-Soaked committed:
Merge branch 'techfab' of https://github.com/MemedHams/Shiptest into techfab

Sun-Soaked committed:
swaw

Sun-Soaked commented:
what a goofy situation I got myself into with this one. Alright, this should be relatively ready to test when I update the desc and it finishes checks.

Sun-Soaked committed:
whoops!

Sun-Soaked committed:
missing research kits

the techification

Sun-Soaked committed:
ඞ

Sun-Soaked committed:
codersuspriting

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
fixes ship rnd

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked commented:
how many times do I have to teach you this lesson, old man

Sun-Soaked committed:
Merge branch 'master' into techfab

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into techfab

Sun-Soaked committed:
Merge branch 'master' into techfab

MarkSuckerberg reviewed (approved):
Code seems fine, honestly, the amount of files is misleading, there's a lot of single var additions.

PrefabQuasar commented:
the nerf to point reward for dissections on common species is probably not needed, since it makes dissections even more of a chore, and honestly the point rewards for rare corpses were already high enough as is, i don't see much of a point to increasing them further, since it only makes research more reliant on pure RNG

Sun-Soaked commented:
> the nerf to point reward for dissections on common species is probably not needed, since it makes dissections even more of a chore, and honestly the point rewards for rare corpses were already high enough as is, i don't see much of a point to increasing them further, since it only makes research more reliant on pure RNG

This also adds direct redeeming of plants found in the wild for points, as an alternate point generation, and makes surveying semi-viable but less repeatable. Also, the sheer amount of buffed power the rare dissections get massively compensates for the common loss- the point is to discourage people from doing common corpses at all, unless in a particularly compromising situation of some sort. 

If you have no rare corpses, you should be surveying (this is faster as of this pr but can only be done a few times, to reduce old survey's tedium) or looking for wild/non-wild plants to redeem for research points by hitting a console with them.

Sun-Soaked commented:
also, any idea why the checks are frozen on this one?

Kapu1178 reviewed (changes requested):
too late to do an actual review, but, these got deleted
